scripts/clang_format_wrapper.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# clang-format misunderstand sql function written in C because they have the signature
+#   Datum my_func(PG_FUNCTION_ARGS)
+# and clang-format does not interpret PG_FUNCTION_ARGS as a type, name pair.
+# This script replaces PG_FUNCTION_ARGS with "PG_FUNCTION_ARGS fake_var_for_clang" to
+# make it look like a proper function for clang and then converts it back after clang runs.
+
+for opt in "$@"
+do
+    if [[ "${opt:0:1}" != "-" ]]
+    then
+        sed -i '' -e 's/PG_FUNCTION_ARGS/PG_FUNCTION_ARGS fake_var_for_clang/' $opt
+        echo $opt
+    fi
+done
+
+clang-format $@
+
+for opt in "$@"
+do
+    if [[ "${opt:0:1}" != "-" ]]
+    then
+        sed -i '' -e 's/PG_FUNCTION_ARGS fake_var_for_clang/PG_FUNCTION_ARGS/' $opt
```

Comment:
the `''` after -i is not needed

Reply:
It is on macos

Reply:
turns out this doesn't work in a way that's portable between linux and macos either. Needed to make it write to a tmp file